src/Workspaces/Core/Portable/Shared/Utilities/AsyncBatchingWorkQueue`2.cs

```diff
@@ -140,7 +141,19 @@ void AddItemsToBatch(IEnumerable<TItem> items)
             async Task<TResult?> ContinueAfterDelay(Task lastTask)
             {
                 using var _ = _asyncListener.BeginAsyncOperation(nameof(AddWork));
-                await lastTask.ConfigureAwait(false);
+                try
+                {
+                    await lastTask.ConfigureAwait(false);
+                }
+                catch (Exception ex) when (FatalError.ReportAndPropagateUnlessCanceled(ex, _cancellationToken, ErrorSeverity.Critical))
+                {
+                    // Make sure the task being performed doesn't propagate out a cancellation exception that isn't
+                    // associated with the token we pass into it.  We don't want an errant cancellation token for some
+                    // other reason to cancel this queue from performing work.
+                }
```

Comment:
💡 This exception handler targets the `_processBatchAsync` callback, but it's not clear from the current handling location. If this code was moved from the current location into `ProcessNextBatchAsync`, it would be more clear which call needed protection.